Sources/OpenAPIRuntime/Conversion/Converter+Client.swift

```diff
@@ -83,20 +83,25 @@ extension Converter {
     /// - Parameters:
     ///   - type: Type used to decode the data.
     ///   - data: Encoded body data.
-    ///   - transform: Closure for transforming the Decodable type into a final type.
+    ///   - strategy: A hint about which coding strategy to use.
+    ///   - transform: Closure for transforming the Decodable type into a final
+    ///   type.
     /// - Returns: Deserialized body value.
     public func bodyGet<T: Decodable, C>(
         _ type: T.Type,
         from data: Data,
+        strategy: BodyCodingStrategy,
         transforming transform: (T) -> C
     ) throws -> C {
         let decoded: T
-        if let myType = T.self as? _StringParameterConvertible.Type {
+        if let myType = T.self as? _StringParameterConvertible.Type,
+            strategy == .string
+        {
```

Comment:
AFAICT this still has fallback semantics if called with `.string` when `T` does _not_ conform to `_StringParameterConvertible`, which IMO would be a bug in our generated code. As we discussed in the other thread, we think this method should be stricter since we have the more lenient method for backwards compatibility still.